File: app/src/main/java/ssilvalucas/testeandroidv2/screen/login/LoginRouter.java
package ssilvalucas.testeandroidv2.screen.login;

import android.content.Intent;
import java.lang.ref.WeakReference;
import ssilvalucas.testeandroidv2.data.model.UserAccount;
import ssilvalucas.testeandroidv2.screen.home.HomeActivity;

interface LoginRouterInput {
    void startHomeActivity(UserAccount userAccount);
}

public class LoginRouter implements LoginRouterInput {

    public static final String TAG = LoginRouter.class.getSimpleName();

    public WeakReference<LoginActivity> activity;

    @Override
    public void startHomeActivity(UserAccount userAccount) {
        Intent intent = new Intent(activity.get(), HomeActivity.class);
        intent.putExtra("userAccount", userAccount);
        activity.get().startActivity(intent);
        activity.get().finish();
    }
}

File: app/src/main/java/com/farage/testeandroidv2/router/LoginRouter.kt
package com.farage.testeandroidv2.router

import android.content.Intent
import android.os.Bundle
import com.farage.testeandroidv2.domain.model.UserAccount
import com.farage.testeandroidv2.ui.HomeActivity
import com.farage.testeandroidv2.ui.LoginActivity
import com.farage.testeandroidv2.util.Constants
import kotlinx.coroutines.ExperimentalCoroutinesApi

@ExperimentalCoroutinesApi
class LoginRouter {

    fun moveUserToHome(activity: LoginActivity, data: UserAccount?): Intent {
        val intent = Intent(activity, HomeActivity::class.java)
        val bundleData = Bundle()
        bundleData.putSerializable(Constants.USER_KEY_BUNDLE, data)
        intent.putExtras(bundleData)
        return intent
    }

    fun moveUserToLoginPage(activity: HomeActivity): Intent = Intent(activity, LoginActivity::class.java)


}